packages/react/README.md

```diff
@@ -15,3 +15,155 @@
+
+### `useSignTransaction(uiWalletAccount, chain)`
+
+Given a `UiWalletAccount` and a chain that begins with `solana:`, this hook returns a function you can call to sign a serialized transaction.
+
+#### Arguments
+
+A config object with the following properties:
+
+-   `options`: An object with the following properties that govern how the wallet should simulate the transaction for the benefit of the user
+    -   `minContextSlot`: The minimum slot at which to simulate the transaction
+    -   `preflightCommitment`: The commitment at which to simulate the transaction
+-   `transaction`: A `Uint8Array` of bytes that conforms to the [Solana transaction schema](https://solana.com/docs/core/transactions#transaction)
+
+#### Returns
+
+An object with the following properties:
+
+-   `signedTransaction`: A `Uint8Array` of bytes that conforms to the [Solana transaction schema](https://solana.com/docs/core/transactions#transaction)
+
+#### Example
+
+```tsx
+import { useSignTransaction } from '@solana/react';
+
+function SignTransactionButton({ account, transactionBytes }) {
+    const signTransaction = useSignTransaction(account, 'solana:devnet');
+    return (
+        <button
+            onClick={async () => {
+                try {
+                    const { signedTransaction } = await signTransaction({
+                        transaction: transactionBytes,
+                    });
+                    window.alert(`Signed transaction bytes: ${signedTransaction.toString()}`);
+                } catch (e) {
+                    console.error('Failed to sign transaction', e);
+                }
+            }}
+        >
+            Sign Transaction
+        </button>
+    );
+}
+```
+
+> [!NOTE]
+> There exists a plural version of this hook &ndash; `useSignTransactions` &ndash; that accepts multiple inputs and returns an array of outputs.
+
+### `useSignAndSendTransaction(uiWalletAccount, chain)`
+
+Given a `UiWalletAccount` and a chain that begins with `solana:`, this hook returns a function you can call to sign and send a serialized transaction.
+
+#### Arguments
+
+A config object with the following properties:
+
+-   `options`: An object with the following properties that govern how the wallet should simulate the transaction for the benefit of the user
+    -   `commitment`: The commitment at which to confirm the transaction
+    -   `maxRetries`: The number of retries the wallet's RPC should make when forwarding the transaction to the leader.
+    -   `minContextSlot`: The minimum slot at which to simulate the transaction
+    -   `mode`: One of `parallel` or `serial`. It's up to the wallet to interpret how to present this to the user.
+    -   `preflightCommitment`: The commitment at which to simulate the transaction
+    -   `skipPreflight`: When `true` asks the wallet to disable transaction verification when sending the transaction to an RPC.
```

Comment:
We should deprecate and eliminate `commitment`, `maxRetries`, `preflightCommitment`, `skipPreflight`. Wallets should not let apps influence how these are simulated or sent (that's up to them) or confirmed (should always use `commitment` `confirmed` anyway, which is what they do).

`mode` can also go (it's not used, and was the wrong way to handle `signAndSendAllTransactions`), which needs its own feature (and no wallet has it yet).